### PR TITLE
Rename JDBC spans to lower cardinality names

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Normally JDBC instrumentation puts the same value in `db.operation` and span name: full SQL query.

### CODE REVIEW

File: custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    }
+  }
+
+  private boolean isJdbcSpan(AttributeValue dbSystem) {
+    return dbSystem != null
+        && dbSystem.getType() == AttributeValue.Type.STRING
+        && SQL_SYSTEMS.contains(dbSystem.getStringValue());
+  }
+
+  private String getStatementType(String query) {
```
Comment (iNikem) [inline custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java:60]: I think if you change this to `SELECT ` etc, then you can ditch `isJdbcSpan`

File: custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.HashSet;
+import java.util.Set;
+
+public class JdbcSpanRenamingProcessor implements SpanProcessor {
```
Comment (iNikem) [inline custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java:27]: Add javadoc please

File: custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java:58]: Hmm, you mean detecting SQL spans just by the prefix? Hmm, I think it's a bit safer this way - this processor won't rename any random span (not necessarily DB related) whose name begins with `select ` (e.g. span manually added by the instrumented app).

File: custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java:34]: Done.

File: custom/src/main/java/com/splunk/opentelemetry/InstrumentationLibraryTracerCustomizer.java [hunk 3]
```diff
@@ -36,11 +39,22 @@ private static boolean spanProcessorInstrumentationLibraryEnabled() {
     return ("true".equalsIgnoreCase(value));
   }
 
+  private static boolean jdbcSpanLowCardinalityNameEnabled() {
+    String value = System.getProperty(ENABLE_JDBC_SPAN_LOW_CARDINALITY_NAME_PROPERTY);
+    if (value == null) {
+      value = System.getenv(propertyToEnv(ENABLE_JDBC_SPAN_LOW_CARDINALITY_NAME_PROPERTY));
+    }
+    // enabled by default
+    return value == null || "true".equalsIgnoreCase(value);
```
Comment (jkwatson) [inline custom/src/main/java/com/splunk/opentelemetry/InstrumentationLibraryTracerCustomizer.java:48]: How about just `Boolean.valueOf(value)`. I think the logic is the same.

Comment (mateuszrzeszutek) [reply custom/src/main/java/com/splunk/opentelemetry/InstrumentationLibraryTracerCustomizer.java:48]: Pretty close, but unfortunately not the same:

```java
    public static Boolean valueOf(String s) {
        return parseBoolean(s) ? TRUE : FALSE;
    }

    public static boolean parseBoolean(String s) {
        return "true".equalsIgnoreCase(s);
    }
```

Comment (jkwatson) [reply custom/src/main/java/com/splunk/opentelemetry/InstrumentationLibraryTracerCustomizer.java:48]: so the difference is? From the javadoc, it looks like the same thing to me.

Comment (mateuszrzeszutek) [reply custom/src/main/java/com/splunk/opentelemetry/InstrumentationLibraryTracerCustomizer.java:48]: Oh, I just realized that you meant just the  `"true".equalsIgnoreCase(value)` part, not the entire line 🤦‍♂️

Comment (mateuszrzeszutek) [reply custom/src/main/java/com/splunk/opentelemetry/InstrumentationLibraryTracerCustomizer.java:48]: In that case, yes, you're exactly right -- I'll change that.

Comment (jkwatson) [reply custom/src/main/java/com/splunk/opentelemetry/InstrumentationLibraryTracerCustomizer.java:48]: right.  :). sorry for the confusion.

File: custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java (hunk 1 above)
Comment (iNikem) [reply custom/src/main/java/com/splunk/opentelemetry/JdbcSpanRenamingProcessor.java:65]: Ok, makes sense. I just did not like this hard-copied list of db systems :(

**Review by iNikem — APPROVED** [review]

**Pull request merged**